insights/parsers/nova_user_ids.py

```diff
@@ -0,0 +1,79 @@
+"""
+Get ``uid`` of user ``nova`` and ``nova_migration``
+=====================================================
+
+The parser class in this module uses base parser class
+``CommandParser`` to get uids of the user ``nova`` and
+``nova_migration``.
+
+Parsers included in this module are:
+
+NovaUID - command ``id -u nova``
+--------------------------------
+
+NovaMigrationUID - command ``id -u nova_migration``
+----------------------------------------------------
+"""
+from insights import CommandParser, parser
+from insights.parsers import ParseException
+from insights.specs import Specs
+
+
+@parser(Specs.nova_uid)
+class NovaUID(CommandParser):
+    '''Parse output of ``id -u nova`` and get the ``uid`` (int).
+    Return ``None`` if nova user not found.
+
+    Typical output of the ``id -u nova`` command is::
+
+        162
+
+    However the id number may vary.
+
+    Examples:
+
+        >>> nova_uid.data
+        162
+        >>> nova_user_not_found.data is None
+        True
+
+    Attributes:
+        data: ``int`` if nova user exist otherwise ``None``.
+
+    '''
+    def parse_content(self, content):
+        self.data = None
+        if not content:
+            raise ParseException("Input content is empty.")
+        if len(content) == 1 and "no such user" not in content[0]:
+            self.data = int(content[0])
+
+
+@parser(Specs.nova_migration_uid)
+class NovaMigrationUID(CommandParser):
+    '''Parse output of ``id -u nova_migration`` and get the ``uid`` (int).
+    Return ``None`` if nova_migration user not found.
+
+    Typical output of the ``id -u nova_migration`` command is::
+
+        153
+
+    However the id number may vary.
+
+    Examples:
+
+        >>> nova_migration_uid.data
+        153
+        >>> nova_migration_user_not_found.data is None
+        True
+
+    Attributes:
+        data: ``int`` if nova_migration user exist otherwise ``None``.
+
+    '''
+    def parse_content(self, content):
+        self.data = None
+        if not content:
+            raise ParseException("Input content is empty.")
+        if len(content) == 1 and "no such user" not in content[0]:
```

Comment:
How about raising a `SkipException` when `no such user`, to avoid generating a parser object which you have to check if it's None in rules?

Reply:
@xiangce As of now I don't see much difference in the functionality of `ParseException` and `SkipException`. However yous suggestion is valid and `SkipException` is more suitable in the current scenario. I'll implement the changes.

Reply:
Done

Reply:
@psachin `SkipException` is appropriate when the parser recognizes that there is no data to process.  `ParseException` should be raised when an error occurs in the data that is unexpected.  `ParseException` is logged and should be investigated for errors in insights-core or data issues.  `SkipException` is ignored by the engine.

If it is important to your rule to know that there is no `nova` user on the system then you should write a separate parser that looks explicitly for that condition.  

In this case I agree with @xiangce that you could include the check for "no such user" as part of your logic to raise the `SkipException`, for example:

```
if not content or (len(content) == 1 and "no such user" in content[0]):
    raise SkipException()
```
Please apply this to the other parser as well.